pkg/sdk/agent_test.go

```diff
@@ -421,3 +421,49 @@ func generateKeys(t *testing.T, keyType string) (priv any, pub []byte) {
 		return privKey, pubKeyBytes
 	}
 }
+
+func TestAlgoError(t *testing.T) {
```

Comment:
This is only a single case. We need `TestAlgo` and add multiple test cases in a single test.